keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -223,9 +228,10 @@ def call_with_cache(self, token_ids, cache, cache_index):
                 whole sequence.
 
         Returns:
-            A (logits, cache) tuple. Where the first output is the language
-            model logits for the input token_ids and the second output is the
-            cache.
+            A (logits, hidden_states, cache) tuple. Where `logits` is the
+            language model logits for the input token_ids, `hidden_states` is
+            the hidden state of the input (the layer before embedding matrix
```

Comment:
Instead of "the hidden state of the input (the layer before embedding matrix mapping)" I would instead say

"the final hidden representation of the input tokens"